src/python/pants/option/options.py

```diff
@@ -346,7 +347,7 @@ def get_fingerprintable_for_scope(self, bottom_scope, include_passthru=False,
       for (_, kwargs) in sorted(parser.option_registrations_iter()):
         if kwargs.get('recursive', False) and not kwargs.get('recursive_root', False):
           continue  # We only need to fprint recursive options once.
-        if bool(invert) == bool(kwargs.get(fingerprint_key, False)):
+        if kwargs.get(fingerprint_key, fingerprint_default) is not True:
```

Comment:
I think this is >10x more readable and will hopefully avoid future confusion about the `invert` argument.